sync/src/main/java/tech/pegasys/teku/sync/historical/HistoricalBlockSyncService.java

```diff
@@ -0,0 +1,269 @@
+
+/**
+ * Service responsible for syncing missing historical blocks. Blocks are pulled in order from the
+ * newest unknown block back to genesis.
+ */
+public class HistoricalBlockSyncService extends Service {
+  private static final Logger LOG = LogManager.getLogger();
+
+  private static final Duration RETRY_TIMEOUT = Duration.ofMinutes(1);
+  private static final UInt64 BATCH_SIZE = UInt64.valueOf(50);
+
+  private final SettableGauge historicSyncGauge;
+  private final StorageUpdateChannel storageUpdateChannel;
+  private final AsyncRunner asyncRunner;
+  private final P2PNetwork<Eth2Peer> network;
+  private final CombinedChainDataClient chainData;
+  private final SyncStateProvider syncStateProvider;
+  private final UInt64 batchSize;
+
+  private final AtomicLong syncStateSubscription = new AtomicLong(-1);
+  private final AtomicBoolean requestInProgress = new AtomicBoolean(false);
+
+  private volatile BeaconBlockSummary earliestBlock;
+  final Set<NodeId> badPeerCache;
+
+  @VisibleForTesting
+  HistoricalBlockSyncService(
+      final MetricsSystem metricsSystem,
+      final StorageUpdateChannel storageUpdateChannel,
+      final AsyncRunner asyncRunner,
+      final P2PNetwork<Eth2Peer> network,
+      final CombinedChainDataClient chainData,
+      final SyncStateProvider syncStateProvider,
+      final UInt64 batchSize) {
+    this.storageUpdateChannel = storageUpdateChannel;
+
+    this.asyncRunner = asyncRunner;
+    this.network = network;
+    this.chainData = chainData;
+    this.syncStateProvider = syncStateProvider;
+    this.batchSize = batchSize;
+
+    this.badPeerCache =
+        Collections.newSetFromMap(
+            CacheBuilder.newBuilder()
+                .maximumSize(100)
+                .expireAfterWrite(Duration.ofMinutes(5))
+                .removalListener(__ -> logBadPeerCacheSize(false))
+                .<NodeId, Boolean>build()
+                .asMap());
+
+    this.historicSyncGauge =
+        SettableGauge.create(
+            metricsSystem,
+            TekuMetricCategory.BEACON,
+            "historical_block_sync_earliest_block",
+            "The slot of the earliest block retrieved by the historical block sync service");
+  }
+
+  public static HistoricalBlockSyncService create(
+      final MetricsSystem metricsSystem,
+      final StorageUpdateChannel storageUpdateChannel,
+      final AsyncRunner asyncRunner,
+      final P2PNetwork<Eth2Peer> network,
+      final CombinedChainDataClient chainData,
+      final SyncStateProvider syncStateProvider) {
+    return new HistoricalBlockSyncService(
+        metricsSystem,
+        storageUpdateChannel,
+        asyncRunner,
+        network,
+        chainData,
+        syncStateProvider,
+        BATCH_SIZE);
+  }
+
+  @Override
+  protected SafeFuture<?> doStart() {
+    LOG.debug("Start {}", getClass().getSimpleName());
+    return initialize().thenAccept(__ -> requestBlocksIfAppropriate());
```

Comment:
May be better that way? 
```suggestion
    return initialize().thenRun(this::requestBlocksIfAppropriate);
```